trino/client.py

```diff
@@ -906,67 +913,138 @@ def _double_map_func(self):
             else NAN if val == 'NaN' \
             else float(val)
 
-    def _timestamp_map_func(self, column, col_type):
-        datetime_default_size = 20  # size of 'YYYY-MM-DD HH:MM:SS.' (the datetime string up to the milliseconds)
-        pattern = "%Y-%m-%d %H:%M:%S"
-        ms_size, ms_to_trim = self._get_number_of_digits(column)
-        if ms_size > 0:
-            pattern += ".%f"
 
-        dt_size = datetime_default_size + ms_size - ms_to_trim
-        dt_tz_offset = datetime_default_size + ms_size
-        if 'with time zone' in col_type:
+class AbstractTemporalRowMapperFactory:
+    def _get_number_of_millis_digits(self, column) -> int:
+        args = column['arguments']
+        if len(args) == 0:
+            return 3
+        ms_size = column['arguments'][0]['value']
+        if ms_size == 0:
+            return -1
+        return ms_size
 
-            if ms_to_trim > 0:
-                return lambda val: \
-                    [datetime.strptime(val[:dt_size] + val[dt_tz_offset:], pattern + ' %z')
-                        if tz.startswith('+') or tz.startswith('-')
-                        else datetime.strptime(dt[:dt_size] + dt[dt_tz_offset:], pattern)
-                                     .replace(tzinfo=pytz.timezone(tz))
-                        for dt, tz in [val.rsplit(' ', 1)]][0]
-            else:
-                return lambda val: [datetime.strptime(val, pattern + ' %z')
-                                    if tz.startswith('+') or tz.startswith('-')
-                                    else datetime.strptime(dt, pattern).replace(tzinfo=pytz.timezone(tz))
-                                    for dt, tz in [val.rsplit(' ', 1)]][0]
+    def _get_number_of_millis_digits_to_trim(self, column, number_millis_digits) -> int:
+        args = column['arguments']
+        if len(args) == 0:
+            return 1
+        return (10 ** (number_millis_digits - min(number_millis_digits, 6)))
 
-        if ms_to_trim > 0:
-            return lambda val: datetime.strptime(val[:dt_size] + val[dt_tz_offset:], pattern)
-        else:
-            return lambda val: datetime.strptime(val, pattern)
 
-    def _time_map_func(self, column, col_type):
-        pattern = "%H:%M:%S"
-        ms_size, ms_to_trim = self._get_number_of_digits(column)
-        if ms_size > 0:
-            pattern += ".%f"
+class TimeRowMapperFactory(AbstractTemporalRowMapperFactory):
```

Comment:
This is actually a value mapper and not a row mapper. Please adjust the name

Reply:
Fixed